### PR TITLE
Release Notes for version 2.8

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Release Notes for version 2.8
 
### Issues Resolved
Resolves #[Issue number to be closed when this PR is merged]
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [X ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: release/release-notes/data-prepper.release-notes-2.8.0.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+## 2024-05-15 Version 2.8.0
+
+---
+
+### Features
+* Support conditional expression to evaluate based on the data type for a given field ([#4478](https://github.com/opensearch-project/data-prepper/issues/4478))
```
Comment (dlvenable) [inline release/release-notes/data-prepper.release-notes-2.8.0.md:6]: Combine this with the PRs:

```suggestion
* Support conditional expression to evaluate based on the data type for a given field ([#4478](https://github.com/opensearch-project/data-prepper/issues/4478), [#4523](https://github.com/opensearch-project/data-prepper/pull/4523), [#4500](https://github.com/opensearch-project/data-prepper/pull/4500))
```

File: release/release-notes/data-prepper.release-notes-2.8.0.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+* Better metrics on OpenSearch document errors ([#4344](https://github.com/opensearch-project/data-prepper/issues/4344))
+* Better metrics for OpenSearch duplicate documents ([#4343](https://github.com/opensearch-project/data-prepper/issues/4343))
+* Address route and subpipeline for pipeline tranformation ([#4528](https://github.com/opensearch-project/data-prepper/pull/4528))
+* Add support for BigDecimal in ConvertType processor ([#4316](https://github.com/opensearch-project/data-prepper/issues/4316))
+* Checkpoint records at an interval for TPS case when AckSet is enabled  ([#4526](https://github.com/opensearch-project/data-prepper/pull/4526))
+* Complete AcknowledgementSet for checkpoint thread & fix BsonTimeStamp field conversion ([#4525](https://github.com/opensearch-project/data-prepper/pull/4525))
+* Write stream events that timeout to write to internal buffer in separate thread ([#4524](https://github.com/opensearch-project/data-prepper/pull/4524))
+* Modify typeof operator grammar and add support for ArrayList ([#4523](https://github.com/opensearch-project/data-prepper/pull/4523))
```
Comment (dlvenable) [inline release/release-notes/data-prepper.release-notes-2.8.0.md:31]: We can remove:

```suggestion
```

File: release/release-notes/data-prepper.release-notes-2.8.0.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+* Add logging change to documentdb source and s3 source with folder par… ([#4519](https://github.com/opensearch-project/data-prepper/pull/4519))
+* Add bytesProcessed metric for DocumentDB stream ([#4518](https://github.com/opensearch-project/data-prepper/pull/4518))
+* Exclude updateDescription field from Mongo/DocDB change stream ([#4516](https://github.com/opensearch-project/data-prepper/pull/4516))
+* Fix document version for DocDB/Mongo Stream events ([#4513](https://github.com/opensearch-project/data-prepper/pull/4513))
+* Add metadata for primary key and primary key type and validations for Mongo/DocDb source ([#4512](https://github.com/opensearch-project/data-prepper/pull/4512))
+* Update docdb metrics ([#4508](https://github.com/opensearch-project/data-prepper/pull/4508))
+* Add BsonType of primary key to metadata ([#4506](https://github.com/opensearch-project/data-prepper/pull/4506))
+* Add bucket owner support to s3 sink ([#4504](https://github.com/opensearch-project/data-prepper/pull/4504))
+* Add 'typeof' operator to DataPrepper expressions ([#4500](https://github.com/opensearch-project/data-prepper/pull/4500))
```
Comment (dlvenable) [inline release/release-notes/data-prepper.release-notes-2.8.0.md:41]: We can remove:

```suggestion
```

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: release/release-notes/data-prepper.release-notes-2.8.0.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
+## 2024-05-15 Version 2.8.0
+
+---
+
+### Features
```
Comment (dlvenable) [inline release/release-notes/data-prepper.release-notes-2.8.0.md:5]: Please also add #4340 which we just backported.

**Review by dlvenable — CHANGES_REQUESTED** [review]

**Review by dlvenable — CHANGES_REQUESTED** [review]
There are still a few comments that are open.

File: release/release-notes/data-prepper.release-notes-2.8.0.md [hunk 5]
```diff
@@ -62,7 +59,6 @@
 * [BUG] Empty DLQ entries when version conflicts occur ([#4301](https://github.com/opensearch-project/data-prepper/issues/4301))
 * [BUG] otel sources should show a more clear exception when receiving data that cannot be processed based on the configured compression type ([#4022](https://github.com/opensearch-project/data-prepper/issues/4022))
 * [BUG] : unable to set field_delimiter_regex ([#2946](https://github.com/opensearch-project/data-prepper/issues/2946))
-* [BUG]Incorrect Behavior of Obfuscate Processor with Predefined Pattern "%{CREDIT_CARD_NUMBER}" [#4340](https://github.com/opensearch-project/data-prepper/issues/4340)
```
Comment (dlvenable) [inline release/release-notes/data-prepper.release-notes-2.8.0.md]: This one should stay.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: release/release-notes/data-prepper.release-notes-2.8.0.md (hunk 5 above)
Comment (kkondaka) [reply release/release-notes/data-prepper.release-notes-2.8.0.md]: @dlvenable it is there. I removed a duplicate line.

**Review by dlvenable — APPROVED** [review]

**Pull request merged**